server/chatserver.cpp

```diff
@@ -1,56 +1,41 @@
 #include "chatserver.h"
 
-static QString getIdentifier(QWebSocket* socket)
-{
-    return QStringLiteral("%1:%2").arg(socket->peerAddress().toString(),
-                                       QString::number(socket->peerPort()));
+static QString getIdentifier(QWebSocket* socket) {
+    return QStringLiteral("%1:%2").arg(socket->peerAddress().toString(), QString::number(socket->peerPort()));
 }
 
-ChatServer::ChatServer(quint16 port, QObject *parent)
+ChatServer::ChatServer(quint16 port, QObject* parent)
     : QObject{parent},
-      m_webSocketServer{new QWebSocketServer("Chat Server",
-                            QWebSocketServer::NonSecureMode,
-                            this)}
-{
-    if(m_webSocketServer->listen(QHostAddress::Any, port))
-    {
+      m_webSocketServer{std::make_unique<QWebSocketServer>("Chat Server", QWebSocketServer::NonSecureMode, this)} {
+    if (m_webSocketServer->listen(QHostAddress::Any, port)) {
         qInfo() << "Server listening on port " << port;
-        connect(m_webSocketServer, &QWebSocketServer::newConnection, this, &ChatServer::onNewConnection);
+        connect(&*m_webSocketServer, &QWebSocketServer::newConnection, this, &ChatServer::onNewConnection);
     }
 }
 
-ChatServer::~ChatServer(){
-    m_webSocketServer->close();
-}
-
-void ChatServer::onNewConnection()
-{
-    auto clientSocket = m_webSocketServer->nextPendingConnection();
-    qInfo() << getIdentifier(clientSocket) << " connected!";
-    clientSocket->setParent(this);
+ChatServer::~ChatServer() { m_webSocketServer->close(); }
 
-    connect(clientSocket, &QWebSocket::textMessageReceived, this, &ChatServer::processMessage);
-    connect(clientSocket, &QWebSocket::disconnected, this, &ChatServer::socketDisconnected);
+void ChatServer::onNewConnection() {
+    auto clientSocket = std::make_unique<QWebSocket*>(m_webSocketServer->nextPendingConnection());
```

Comment:
I think it is a bad idea to create unique_ptr for raw ptr,
maybe try like this:
auto clientSocket = std::make_unique<QWebSocket>(*m_webSocketServer->nextPendingConnection());
?

Reply:
It's not possible, when I do this I got chatserver.cpp:19: error: In template: call to deleted constructor of 'QWebSocket'